packages/python/genai_prices/types.py

```diff
@@ -154,6 +154,44 @@ def __repr__(self) -> str:
             f'auto_update_timestamp={self.auto_update_timestamp!r})'
         )
 
+    def __add__(self, other: ExtractedUsage | Any) -> ExtractedUsage:
+        """Accumulate inner Usage, handling nullable usage fields.
+
+        Accumulating usage is useful for common streaming situations where user wants to save and compute costs for
+        all the response chunks in a stream
+
+        Args:
+              other: The usage to accumulate with this usage extraction instance.
+        """
+
+        if not isinstance(other, ExtractedUsage):
+            raise ValueError(f'Cannot add {other} to {self}, other is not an ExtractedUsage')
+
+        models_match = other.model.id == self.model.id if other.model and self.model else False
+        if not models_match:
+            raise ValueError(f'Cannot add {other} to {self}, models do not match {other.model} != {self.model}')
+
+        def _add_option(a: int | None, b: int | None) -> int | None:
+            return None if all(num is None for num in (a, b)) else (a or 0) + (b or 0)
+
+        updated_usage = Usage(
+            input_tokens=_add_option(self.usage.input_tokens, other.usage.input_tokens),
```

Comment:
might be better to loop over the field names so that this automatically works if in future we add more fields like image tokens

if not, then all token types should be covered in the tests